gpflow/models/model.py

```diff
@@ -44,8 +44,12 @@ def neg_log_marginal_likelihood(self, *args, **kwargs) -> tf.Tensor:
     def log_marginal_likelihood(self, *args, **kwargs) -> tf.Tensor:
         return self.log_likelihood(*args, **kwargs) + self.log_prior()
 
-    def log_prior(self) -> tf.Tensor:
-        log_priors = [p.log_prior() for p in self.trainable_parameters]
+    def log_marginal_likelihood_on_unconstrained(self, *args, **kwargs) -> tf.Tensor:
+        return self.log_likelihood(*args, **kwargs) + self.log_prior(evaluate_on_constrained=False)
```

Comment:
Why not have this as an argument to `log_marginal_likelihood`?

Reply:
Done, thanks!